### PR TITLE
Add a Validation To Assure Feed Integrations Have A Reliability Parameter

### PR DESCRIPTION
**Pull request by MichaelYochpaz**
## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-3678

## Description
Make the existing `IN154` run on non-modified (new) files as well, and assure feeds have a reliability parameter as well.

### CODE REVIEW

File: demisto_sdk/commands/common/errors.py [hunk 1]
```diff
@@ -2361,13 +2363,22 @@ def invalid_defaultvalue_for_checkbox_field(name: str):
 
     @staticmethod
     @error_code_decorator
-    def missing_reliability_parameter(command: str):
-        return (
-            f'Missing "Reliability" parameter in the {command} reputation command.'
-            f"Please add it to the YAML file."
-            f"For more information, refer to https://xsoar.pan.dev/docs/integrations/dbot#reliability-level"
+    def missing_reliability_parameter(is_feed: bool, command_name: str | None = None):
+        # Assure function is called properly, as 'command' is required when 'is_feed' is False
```
Comment (bziser) [inline demisto_sdk/commands/common/errors.py:2367]: please add Docstring

File: demisto_sdk/commands/common/errors.py [hunk 2]
```diff
@@ -2361,13 +2363,22 @@ def invalid_defaultvalue_for_checkbox_field(name: str):
...
+        if not is_feed and not command_name:
+            raise ValueError("If is_feed is False, command must be provided.")
+
+        error_message = (
+            "must implement a reliability configuration parameter in the YAML file.\n"
+            "For more information, refer to https://xsoar.pan.dev/docs/integrations/dbot#reliability-level"
```
[the remark this answers is not in the document]
Comment (bziser) [reply demisto_sdk/commands/common/errors.py:2373]: I think the previous was more explicit
We need to say which command and what is the parmater

**Review by bziser — COMMENTED** [review]
great

File: demisto_sdk/commands/common/errors.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (MichaelYochpaz) [reply demisto_sdk/commands/common/errors.py:2373]: Regarding the suggested change, this string is appended to one of 2 other strings, so it ends up being:

>Feed integrations must implement a reliability configuration parameter in the YAML file.
>For more information, refer to https://xsoar.pan.dev/docs/integrations/dbot#reliability-level

or

>Integrations that contain reputation commands ({command_name}) must implement a reliability configuration parameter in the YAML file.
>For more information, refer to https://xsoar.pan.dev/docs/integrations/dbot#reliability-level

**Comment by bziser** [comment]
Adding @AradCarmi also to the review since he wrote the previous validation

File: demisto_sdk/commands/common/errors.py [hunk 3]
```diff
@@ -2361,13 +2364,34 @@ def invalid_defaultvalue_for_checkbox_field(name: str):
...
+            f"in the YAML file.\n"
+            "For more information, refer to https://xsoar.pan.dev/docs/integrations/dbot#reliability-level"
         )
 
+        if is_feed:
+            return f"Feed integrations {error_message}"
+
+        else:
+            return f"Integrations that contain reputation commands ({command_name}) {error_message}"
+
```
Comment (AradCarmi) [inline demisto_sdk/commands/common/errors.py:2394]: I don't think we need to return a different error message.

File: CHANGELOG.md [hunk 4]
```diff
@@ -19,6 +19,7 @@
 * Fixed an issue in **pre-commit** where `--input` flag was not filtered by the git files.
 * Skip reset containers for XSOAR NG and XSIAM(PANW-internal only).
 * Fixed an issue where **lint** failed fetching docker image details from a PANW GitLab CI environment. (PANW-internal only).
+* Added a validation that feed integrations implement the `integrationReliability` configuration parameter.
```
Comment (AradCarmi) [inline CHANGELOG.md:22]:
```suggestion
* Updated the **validate** command to verify feed integrations implement the *integrationReliability* configuration parameter.
```

**Review by AradCarmi — COMMENTED** [review]
Nicely done!
Why do we need to verify the reliability param on feed integration?
The reason that we added it first was when running a reputation command having several instances that implement the command, the server is calculating the indicator verdict between the among instances, depending on each instance reliability param. Why does it matter for feed integration? 
Please also mention the PR that fix/adding the reliability param to the currently missing integration.

File: demisto_sdk/commands/common/errors.py (hunk 3 above)
Comment (MichaelYochpaz) [reply demisto_sdk/commands/common/errors.py:2394]: The validation (after this update) is raised when the reliability parameter is missing in one of two cases (in which the field is required):

1. Integration is a feed (doesn't necessarily have reputation commands)
2. Integration has reputation commands (isn't necessarily a feed)

I think it's better to have separate errors, to make it more clear why the validation failed.

**Comment by MichaelYochpaz** [comment]
@AradCarmi take a look at the linked ticket.
It's a task to implement a validation that assures all feed integrations will have a reputation parameter (no feed integrations failed after the update, so it seems like all of them already do. We're just adding a validation for that)

As for the PR fixing the existing parameters, there isn't one yet, currently working on it (you can see in the `validate-files` step which packs are failing)

**Comment by MichaelYochpaz** [comment]
Content PR for fixing (ignoring) failing validations:
https://github.com/demisto/content/pull/26665

File: demisto_sdk/commands/common/errors.py [hunk 5]
```diff
@@ -2361,13 +2364,34 @@ def invalid_defaultvalue_for_checkbox_field(name: str):
...
+            str: The error message.
+        """
+        # Assure 'command' parameter was passed if 'is_feed' is False
+        if not is_feed and not command_name:
+            raise ValueError("If 'is_feed' is set to False, command must be provided.")
+
+        error_message = (
+            f"must implement a reliability ('{RELIABILITY_PARAMETER_NAMES[0]}') configuration parameter"
+            f"in the YAML file.\n"
+            "For more information, refer to https://xsoar.pan.dev/docs/integrations/dbot#reliability-level"
```
Comment (bziser) [inline demisto_sdk/commands/common/errors.py:2386]:
```suggestion
           f"Missing '{RELIABILITY_PARAMETER_NAMES[0]}')  parameter in the {command_name} command.\n Please add it to the YAML file.\n For more information, refer to https://xsoar.pan.dev/docs/integrations/dbot#reliability-level"
```

File: demisto_sdk/commands/common/errors.py [hunk 6]
```diff
@@ -2361,13 +2364,34 @@ def invalid_defaultvalue_for_checkbox_field(name: str):
...
+                Defaults to None. Required when is_feed is False.
+
+        Returns:
+            str: The error message.
+        """
+        # Assure 'command' parameter was passed if 'is_feed' is False
+        if not is_feed and not command_name:
+            raise ValueError("If 'is_feed' is set to False, command must be provided.")
+
+        error_message = (
```
Comment (bziser) [inline demisto_sdk/commands/common/errors.py:2383]:
```suggestion
        error_message_suffix = (
```

File: demisto_sdk/commands/common/errors.py [hunk 7]
```diff
@@ -2361,11 +2364,37 @@ def invalid_defaultvalue_for_checkbox_field(name: str):
...
+            is_feed (bool): Whether the integration is a feed integration or not.
+            command_name (str | None, optional): The name of the command that is missing the reliability parameter.
+                Defaults to None. Required when is_feed is False.
+
+        Returns:
+            str: The error message.
+        """
+        # Assure 'command' parameter was passed if 'is_feed' is False
+        if not is_feed and not command_name:
+            raise ValueError("If 'is_feed' is set to False, command must be provided.")
```
Comment (YuvHayun) [inline demisto_sdk/commands/common/errors.py:2381]: It's not customary to raise errors for developers here, let's add this note to the function docs.

**Review by YuvHayun — DISMISSED** [review]
Looks good!
1 small comment.

File: demisto_sdk/commands/common/errors.py [hunk 8]
```diff
@@ -2361,11 +2366,37 @@ def invalid_defaultvalue_for_checkbox_field(name: str):
...
+        """
+        if is_feed:
+            specific_case_error = (
+                "Feed integrations must implement a reliability parameter."
+            )
+
+        else:
+            specific_case_error = (
+                "Integrations with reputation commands{0} ".format(
+                    f" ('{command_name}')"  # Add reputation command's name to error if provided
```
Comment (bziser) [inline demisto_sdk/commands/common/errors.py:2389]: I don't think this comment is needed, it is understandable from the code

Comment (MichaelYochpaz) [reply demisto_sdk/commands/common/errors.py:2389]:
```suggestion
                    f" ('{command_name}')"
```

**Review by bziser — APPROVED** [review]

**Pull request merged**